src/Language/PureScript/Interactive/Types.hs

```diff
@@ -57,27 +62,35 @@ data PSCiState = PSCiState
   [ImportedModule]
   [P.Declaration]
   [(P.Module, P.ExternsFile)]
+  (P.ModuleName, P.Ident)
```

Comment:
Neither here nor there, but why not use record syntax for this rather than writing hand written selectors?

Reply:
I suspect this is the way it is because we don’t want to allow updating the fields directly but we do want to allow reading them. Come to think of it we could do something like declaring this as a record with fields prefixed with an underscore eg _psciImportedModules, and not expose the fields but continue to expose the functions (which only allow reading, not updating).